### ISSUE TITLE
Bugs in the delete-current-word gesture

### ISSUE THREAD

**Issue by Glitchy-Tozier** [florisboard/florisboard#525]
(Tested by placing the delete-current-word gesture on space-upward-swipe)

1. When deleting the last word of any text, Floisboard crashes. (See attached error report)
2. The behaviour when deleting a word in the middle of a text is just... jank. No idea what is happening. Things jump around, the next word is deleted alongside the current one, etc. Try it out, it's fun.

#### Environment information
- FlorisBoard Version: 0.3.9
- Install Source: Google PlayStore
- Device: Samsung Galaxy S9
- Android version, ROM: 10

#### Error Report:
~~~ 1617024869194.stacktrace ~~~

java.lang.IndexOutOfBoundsException: Empty list doesn't contain element at index -1.
	at kotlin.collections.EmptyList.get(Collections.kt:36)
	at kotlin.collections.EmptyList.get(Collections.kt:24)
	at dev.patrickgold.florisboard.ime.core.EditorInstance.deleteWordsBeforeCursor(EditorInstance.kt:235)
	at dev.patrickgold.florisboard.ime.text.TextInputManager.handleDeleteWord(TextInputManager.kt:468)
	at dev.patrickgold.florisboard.ime.text.TextInputManager.onInputKeyUp(TextInputManager.kt:704)
	at dev.patrickgold.florisboard.ime.core.InputEventDispatcher$1$3.invokeSuspend(InputEventDispatcher.kt:125)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:8167)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)

**Comment by patrickgold** [comment]
Thanks for your bug report! This crash log (and variants of it) is pretty known and I've already added an potential fix to v0.3.10-beta05 (which will be released this weekend), which will hopefully fix this bug. Closing this issue as the original issue for this is #484.

Regarding the jankiness: Yup, it is jank af :) Will see when I will fix it (currently I am a bit overwhelmed with the amount of new issues and the stuff I plan to implement/am implementing, so sorry if I am responding to your issues a bit late sometimes)

**Comment by Glitchy-Tozier** [comment]
I see, didn't know that's what "precise word deletion" was referring to.

> Regarding the jankiness: Yup, it is jank af :) Will see when I will fix it

Should I create a new issue for that problem? :)

> Will see when I will fix it (currently I am a bit overwhelmed with the amount of new issues and the stuff I plan to implement/am implementing, so sorry if I am responding to your issues a bit late sometimes)

No problem, I can imagine that it's a lot to handle. Tackle everything at a pace that suits you.  :) 
(Although I love your practice of replying within a reasonable time-frame, just know that you can take your time replying to my issues)
